exasol_advanced_analytics_framework/udf_communication/background_peer_state.py

```diff
@@ -0,0 +1,94 @@
+import contextlib
+import time
+from typing import Optional, Generator, List
+
+import zmq
+from zmq import Frame
+
+from exasol_advanced_analytics_framework.udf_communication.connection_info import ConnectionInfo
+from exasol_advanced_analytics_framework.udf_communication.get_peer_receive_socket_name import \
+    get_peer_receive_socket_name
+from exasol_advanced_analytics_framework.udf_communication.messages import Message, WeAreReadyToReceiveMessage, \
+    AreYouReadyToReceiveMessage, PeerIsReadyToReceiveMessage, PayloadMessage
+from exasol_advanced_analytics_framework.udf_communication.peer import Peer
+from exasol_advanced_analytics_framework.udf_communication.serialization import serialize_message
+
+
+class BackgroundPeerState:
+
+    def __init__(self,
+                 my_connection_info: ConnectionInfo,
+                 out_control_socket: zmq.Socket,
+                 context: zmq.Context,
+                 peer: Peer,
+                 reminder_timeout_in_seconds: float = 1):
+        self._out_control_socket = out_control_socket
+        self._my_connection_info = my_connection_info
+        self._wait_time_between_reminder_in_seconds = reminder_timeout_in_seconds
+        self._peer = peer
+        self._context = context
+        self._peer_can_receive_from_us = False
+        self._last_send_ready_to_receive_timestamp_in_seconds: Optional[float] = None
+        self._create_receive_socket()
+        self._send_we_are_ready_to_receive()
+
+    def _create_receive_socket(self):
+        self._receive_socket = self._context.socket(zmq.PAIR)
+        receive_socket_address = get_peer_receive_socket_name(self._peer)
+        self._receive_socket.bind(receive_socket_address)
```

Comment:
that name is actually a bit confusing. As far as I understood that is the data socket between frontend and backend for a specific peer. Maybe rename it to "frontend_data_socket"

Reply:
I now what you mean, however I choose this name, because the frontend will only use this socket to receive message from a specific peer. Here, we will send to the socket while forwarding the payload messages to the frontend.

Reply:
Ok, that's fine.